x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/physical/GrokExec.java

```diff
@@ -90,7 +91,7 @@ public int hashCode() {
         return Objects.hash(super.hashCode(), parser);
     }
 
-    public Grok.Parser pattern() {
+    public Grok.Parser getParser() {
```

Comment:
I'd reduce the scope of this PR to the bare minimum, that is quite big already ;-)